File: qe-selenium-java/src/main/java/demos/ActivitySixteen.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

import java.util.List;

public class ActivitySixteen {
    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();
        driver.get("https://training-support.net/webelements/selects");
        driver.manage().window().maximize();

        System.out.println(driver.getTitle());

        WebElement dropdown = driver.findElement(By.tagName("select"));

        Select select = new Select(dropdown);

        select.selectByVisibleText("One");

        select.selectByIndex(2);

        select.selectByValue("three");

        List<WebElement> options = select.getOptions();

        System.out.println("All dropdown options:");
        for (WebElement option : options) {
            System.out.println(option.getText());
        }

        driver.quit();
    }
}

File: qe-selenium-python/Activity_Sixteen.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import Select

driver = webdriver.Firefox()

driver.get("https://training-support.net/webelements/selects")
driver.maximize_window()

print(driver.title)

dropdown = driver.find_element(By.TAG_NAME, "select")
select = Select(dropdown)

select.select_by_visible_text("One")
select.select_by_index(2)
select.select_by_value("three")

print("All dropdown options:")
for option in select.options:
    print(option.text)

driver.quit()
